src/Swarm/Game/Step.hs

```diff
@@ -419,7 +419,9 @@ entityAt loc = zoomWorld (W.lookupEntityM @Int (W.locToCoords loc))
 -- | Modify the entity (if any) at a given location.
 updateEntityAt ::
   (Has (State GameState) sig m) => Location -> (Maybe Entity -> Maybe Entity) -> m ()
-updateEntityAt loc upd = zoomWorld (W.updateM @Int (W.locToCoords loc) upd)
+updateEntityAt loc upd = do
+  zoomWorld (W.updateM @Int (W.locToCoords loc) upd)
+  wakeWatchingRobots loc
```

Comment:
It's not ideal that I have to call `wakeWatchingRobots` both here (in `updateEntityAt`) and inside the `updateWorld` function.  I wonder if there is consolidation that could be done.  My first question is whether specifically `W.updateM` is needed here and specifically `W.update` is needed in `updateWorld`, or if both places could use the same.

Reply:
I think probably `updateWorld` should be refactored to call `updateEntityAt`.  I don't know if there's a specific reason why `updateWorld` calls `update` instead of `updateM`.  I can't think of any reason why it would ever be important to call `update` specifically.  It's always safe to call `updateM` but perhaps sometimes calling `update` could be an optimization if you know the tile must already be loaded.